src/main/java/com/azure/cosmos/kafka/connect/sink/BulkWriter.java

```diff
@@ -56,15 +54,38 @@ protected SinkWriteResponse writeCore(List<SinkRecord> sinkRecords) {
         }
 
         List<CosmosItemOperation> itemOperations = new ArrayList<>();
-        for (SinkRecord sinkRecord : sinkRecords) {
-            CosmosItemOperation cosmosItemOperation = CosmosBulkOperations.getUpsertItemOperation(
-                    sinkRecord.value(),
-                    this.getPartitionKeyValue(sinkRecord.value()),
-                    new SinkOperationContext(sinkRecord));
+        if (this.noDuplicates) {
+            HashMap<IdAndPartionKey, SinkRecord> uniqueItems = new HashMap<>();
```

Comment:
Do we need a concurrent hash map here?

Reply:
I don't think a concurrent hash map is needed because it should be contained in only one thread because my understanding is that there is one thread per CosmosDBSinkTask.